tests/test_transducers/test_transducers/test_treduce.py

```diff
@@ -0,0 +1,9 @@
+import pytest
+
+from returns.transducers import treduce
+
+
+def test_reduce():
+    """Should fail when iterable is empty and non initial value is given."""
```

Comment:
Should we? Raising exceptions is not something we do in this library (at least we try to).

Can we just return the empty sequence? Or do anything else?